stripe/src/main/java/com/stripe/android/PaymentSession.java

```diff
@@ -0,0 +1,204 @@
+     * occurs. Failure can only occur if there is no initialized {@link CustomerSession}.
+     */
+    public boolean init(@NonNull PaymentSessionListener listener) {
+        return init(listener, null);
+    }
+
+    /**
+     * Initialize the PaymentSession with a {@link PaymentSessionListener} to be notified of
+     * data changes.
+     *
+     * @param listener a {@link PaymentSessionListener} that will receive notifications of changes
+     *                 in payment session status, including networking status
+     * @param savedInstanceState a {@link Bundle} containing the saved state of a PaymentSession
+     *                           that was stored in {@link #savePaymentSessionInstanceState(Bundle)}
+     * @return {@code true} if the PaymentSession is initialized, {@code false} if a state error
+     * occurs. Failure can only occur if there is no initialized {@link CustomerSession}.
+     */
+    public boolean init(
+            @NonNull PaymentSessionListener listener,
+            @Nullable Bundle savedInstanceState) {
+
+        // Checking to make sure that there is a valid CustomerSession -- the getInstance() call
+        // will throw a runtime exception if none is ready.
+        try {
+            CustomerSession.getInstance().addProductUsageTokenIfValid(TOKEN_PAYMENT_SESSION);
+        } catch (IllegalStateException illegalState) {
+            mPaymentSessionListener = null;
+            return false;
+        }
+
+        mPaymentSessionListener = listener;
+        if (savedInstanceState != null) {
+            PaymentSessionData data =
+                    savedInstanceState.getParcelable(PAYMENT_SESSION_DATA_KEY);
+            if (data != null) {
+                mPaymentSessionData = data;
+            }
+        }
+        fetchCustomer();
+        return true;
+    }
+
+    /**
+     * Launch the {@link PaymentMethodsActivity} to allow the user to select a payment method,
+     * or to add a new one.
+     */
+    public void selectPaymentMethod() {
+        mHostActivity.startActivityForResult(
+                PaymentMethodsActivity.newIntent(mHostActivity),
+                PAYMENT_METHOD_REQUEST);
+    }
+
+    /**
+     * Save the data associated with this PaymentSession. This should be called in the host Activity
+     * {@link Activity#onSaveInstanceState(Bundle)} method.
+     *
+     * @param outState the host activity's outgoing {@link Bundle}
+     */
+    public void savePaymentSessionInstanceState(@NonNull Bundle outState) {
+        outState.putParcelable(PAYMENT_SESSION_DATA_KEY, mPaymentSessionData);
+    }
+
+    /**
+     * Set the cart total for this PaymentSession. This should not include shipping costs.
+     *
+     * @param cartTotal the current total price for all non-shipping and non-tax items in
+     *                  a customer's cart
+     */
+    public void setCartTotal(@IntRange(from = 0) long cartTotal) {
```

Comment:
I think we should align on naming and behavior here to match the iOS SDK:
   - We call `cartTotal` -> `paymentAmount`. `paymentAmount` is a value that you set initially with the amount you want to charge. This could include shipping costs (e.g. if you aren't using our shipping UI) or other fees. 
   - We also have a `paymentCurrency` (payment methods like Apple Pay require passing a currency, and it's also necessary for displaying amounts in the shipping UI).
   - If you set a `shippingMethod` (which has `amount`, `identifier`, `label`, and `detail` fields) , `paymentAmount` will update to include the shipping method's amount
   - We don't mention tax, and don't include any tax related APIs

cc @bdorfman-stripe @joeydong-stripe

Reply:
Will do; this is just a skeleton to work on, so we can make these updates.